File: dev/20190402/k2works/sam-app/HelloWorldFunction/src/main/java/helloworld/FizzBuzz.java
package helloworld;


import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class FizzBuzz {
    public static List<String> generateFizzBuzzList() {
        return IntStream.rangeClosed(1,100).mapToObj(FizzBuzz::generate).collect(Collectors.toList());
    }

    public static void print(List<String> list) {
        list.forEach(System.out::println);
    }

    public static String generate(int i) {
        boolean fizz = isFizz(i);
        boolean buzz = isBuzz(i);
        if (fizz && buzz) return "FizzBuzz";
        if (fizz) return "Fizz";
        if (buzz) return "Buzz";
        return Integer.toString(i);
    }

    public static String generate(int i, int type) {
        switch (type) {
            case 1:
                return FizzBuzz.generate(i);
            default:
                return FizzBuzz.generate(i);
        }
    }

    private static boolean isBuzz(int i) {
        return i % 5 == 0;
    }

    private static boolean isFizz(int i) {
        return i % 3 == 0;
    }
}

File: dev/20190902/k2works/app/FizzBuzz/ClientApp/src/domain/FizzBuzz.ts
export class FizzBuzz {
  public static generateList(): any[] {
    const list = [];
    for (let i = 1; i <= 100; i++) {
      list[i - 1] = i;
    }
    return list;
  }

  public static generate(input: number) {
    let result = input.toString();
    if (input % 3 === 0 && input % 5 === 0) {
      result = "FizzBuzz";
    } else if (input % 3 === 0) {
      result = "Fizz";
    } else if (input % 5 === 0) {
      result = "Buzz";
    }
    return result;
  }
}
